galsim/table.py

```diff
@@ -165,18 +179,22 @@ def __call__(self, x):
             if dimen > 2:
                 raise ValueError("Arrays with dimension larger than 2 not allowed!")
             elif dimen == 2:
-                f = np.zeros_like(x)
-                for i in xrange(x.shape[0]):
-                    f[i,:] = np.fromiter((self.table(float(q)) for q in x[i,:]), dtype='float')
+                f = np.empty_like(x.flatten(), dtype=float)
```

Comment:
I noticed a bunch of places in GalSim (either in galsim/ or tests/) where we use ```flatten```.  Should we be switching to ```ravel``` more systematically, for any place where there's no risk of data getting modified?